sklearnex/linear_model/ridge.py

```diff
@@ -14,11 +14,374 @@
+
+            n_samples = _num_samples(X)
+            n_features = _num_features(X, fallback_1d=True)
+
+            # Check if equations are well defined, see LinReg for further explanation
+            is_underdetermined = n_samples < (n_features + int(self.fit_intercept))
+            supports_all_variants = daal_check_version((2025, "P", 1))
+
+            patching_status.and_conditions(
+                [
+                    (
+                        self.solver == "auto",
+                        f"'{self.solver}' solver is not supported. "
+                        "Only 'auto' solver is supported.",
+                    ),
+                    (
+                        not issparse(X) and not issparse(y),
+                        "Sparse input is not supported.",
+                    ),
+                    (
+                        not is_underdetermined or supports_all_variants,
+                        "The shape of X (fitting) does not satisfy oneDAL requirements:"
+                        "Number of features + 1 >= number of samples.",
+                    ),
+                    (sample_weight is None, "Sample weight is not supported."),
+                    (not normalize_is_set, "Normalization is not supported."),
+                    (
+                        not positive_is_set,
+                        "Forced positive coefficients are not supported.",
+                    ),
+                    (
+                        isinstance(self.alpha, numbers.Real),
+                        "Non-scalar alpha is not supported yet.",
+                    ),
+                ]
+            )
+
+            return patching_status
+
+        def _onedal_predict_supported(self, patching_status, method_name, *data):
+            assert method_name in ["predict", "score"]
+            assert len(data) <= 2
+
+            n_samples = _num_samples(data[0])
+            model_is_sparse = issparse(self.coef_) or (
+                self.fit_intercept and issparse(self.intercept_)
+            )
+            patching_status.and_conditions(
+                [
+                    (
+                        self.solver == "auto",
+                        f"'{self.solver}' solver is not supported. "
+                        "Only 'auto' solver is supported.",
+                    ),
+                    (n_samples > 0, "Number of samples is less than 1."),
+                    (not issparse(data[0]), "Sparse input is not supported."),
+                    (not model_is_sparse, "Sparse coefficients are not supported."),
+                ]
+            )
+
+            return patching_status
+
+        def _onedal_gpu_supported(self, method_name, *data):
+            patching_status = PatchingConditionsChain(
+                f"sklearn.linear_model.{self.__class__.__name__}.{method_name}"
+            )
+
+            if method_name == "fit":
+                if isinstance(self.alpha, numbers.Real) and np.isclose(self.alpha, 0):

```

Comment:
Also since the PR that fixes this for GPU was already merged (https://github.com/oneapi-src/oneDAL/pull/2976), perhaps you could also skip this check for version >= 2025.2.